plugins/serializers/prometheus/convert.go

```diff
@@ -55,35 +55,36 @@ var LabelNameTable = Table{
 	},
 }
 
-func isValid(name string, table Table) bool {
-	if name == "" {
+func isValidMetricName(name string) bool {
+	if len(name) == 0 {
 		return false
 	}
-
-	for i, r := range name {
-		switch {
-		case i == 0:
-			if !unicode.In(r, table.First) {
-				return false
-			}
-		default:
-			if !unicode.In(r, table.Rest) {
-				return false
-			}
+	for i := 0; i < len(name); i++ {
+		b := name[i]
+		if !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || b == ':' || (b >= '0' && b <= '9' && i > 0)) {
+			return false
 		}
 	}
```

Comment:
I really would love to see a (pre-compiled) regexp for this. Is there any reason for doing this manual dance?